src/simulation/LoadGenerator.cpp

```diff
@@ -1208,33 +1303,31 @@ LoadGenerator::invokeSorobanLoadTransaction(uint32_t ledgerNum,
     SorobanResources resources;
     resources.footprint.readOnly = instance.readOnlyKeys;
 
-    // Must always read wasm and instance
-    releaseAssert(networkCfg.mTxMaxReadLedgerEntries > 1);
-    auto maxEntries = networkCfg.mTxMaxReadLedgerEntries - 2;
-    maxEntries = std::min(maxEntries, cfg.nDataEntriesHigh);
-    auto minEntries = std::min(maxEntries, cfg.nDataEntriesLow);
-    auto numEntries = rand_uniform<uint32_t>(minEntries, maxEntries);
+    auto numEntries = rand_uniform<uint32_t>(invokeCfg.nDataEntriesLow,
+                                             invokeCfg.nDataEntriesHigh);
     for (uint32_t i = 0; i < numEntries; ++i)
     {
         auto lk = contractDataKey(instance.contractID, makeU32(i),
                                   ContractDataDurability::PERSISTENT);
         resources.footprint.readWrite.emplace_back(lk);
     }
 
-    uint32_t maxKiloBytesPerEntry = 0;
-    if (networkCfg.mTxMaxReadBytes > mContactOverheadBytes && numEntries > 0)
+    auto totalWriteBytes = rand_uniform<uint32_t>(invokeCfg.ioKiloBytesLow,
+                                                  invokeCfg.ioKiloBytesHigh) *
+                           1024;
+
+    uint32_t kiloBytesPerEntry = 0;
+    if (numEntries > 0)
     {
-        maxKiloBytesPerEntry =
-            (networkCfg.mTxMaxReadBytes - mContactOverheadBytes) / numEntries /
-            1024;
-    }
+        kiloBytesPerEntry =
+            (totalWriteBytes - mContactOverheadBytes) / numEntries / 1024;
 
-    maxKiloBytesPerEntry =
-        std::min(maxKiloBytesPerEntry, cfg.kiloBytesPerDataEntryHigh);
-    uint32_t minKiloBytesPerEntry =
-        std::min(maxKiloBytesPerEntry, cfg.kiloBytesPerDataEntryLow);
-    auto kiloBytesPerEntry =
-        rand_uniform<uint32_t>(minKiloBytesPerEntry, maxKiloBytesPerEntry);
+        // If numEntries > 0, we can't write a 0 byte entry
```

Comment:
Hmm, we actually can, can't we? It won't be 0 bytes, probably about 8 bytes to write an empty Bytes ScVal.

Reply:
Sorry, I didn't know this was valid behavior. Currently the loadgen contract panics if `size_kilo_bytes == 0 && num_write_entries != 0`. We can fix this later, but personally I don't think an env change and wasm recompile is worth fixing this right now.

Reply:
Ah, ok. Empty bytes sequence is a valid value for a byte sequence, but I agree there is no need to update Wasm due to that. Also, it doesn't provide any significant value for coverage.